### ISSUE TITLE
Support for doctest

### ISSUE THREAD

**Issue by wilhem-meignan** [matepek/vscode-catch2-test-adapter#143]
**Is your feature request related to a problem? Please describe.**
doctest tests are not recognized by the extension although doctest is based on catch and is very similar. Its main advantage is that it compiles tests much faster that catch2.

**Describe the solution you'd like**
Is it possible to make this extension recognize doctest tests just as well as catch2 tests? Maybe it is just a matter of configuring the extension or doctest.

**Describe alternatives you've considered**
If it is not just a configuration issue, then either a new test adapter extension would be needed or this one would need to be updated to work with doctest. Since doctest is based on catch, this extension would probably be a good starting point in both cases.

**Additional context**
Link to doctest:
https://github.com/onqtam/doctest

**Comment by matepek** [comment]
its a "slightly" bigger change even if the syntax is the same. need time. or you can contribute if you wish?

**Comment by matepek** [comment]
Hey,
I had some time so I put something together. I had no time to write tests and do proper testing, so I call it "experimental".
[vscode-catch2-test-adapter-2.7.9.vsix.zip](https://github.com/matepek/vscode-catch2-test-adapter/files/3946474/vscode-catch2-test-adapter-2.7.9.vsix.zip)
You can install it with a visual studio command "Install from VSIX...".
Let me know if it works or not, bug reports, etc..

**Comment by wilhem-meignan** [comment]
Hi @matepek,

Thanks! This is great!

I have tried it and could only find one issue: when two test cases have the same name they are not run even if they are in separate test suites. In the following example, the test cases are not run:

    TEST_SUITE("First test suite")
    {
        TEST_CASE("one test case")
        {
            REQUIRE(true);
        }
    }

    TEST_SUITE("Second test suite")
    {
        TEST_CASE("one test case")
        {
            REQUIRE(false);
        }
    }

Otherwise, everything I could test worked:
- Detecting test cases. Test targets with no test cases are ignored but that's fine with me.
- Running the tests and displaying the results including the inline display of failed assertions.
- Autorun of test target when it is rebuilt.

**Comment by matepek** [comment]
Happy to hear. 
I’m aware of that issue. But no solution because doctest lacks of a lot of features like proper test name listing. I even have to add some tricks to have code lens. If I have time I will write some feature request or PR to doctest.

**Comment by wilhem-meignan** [comment]
OK. Thank you very much for your work. Do you know when this is going into a release?

**Comment by matepek** [comment]
I can put it now with “experimental” flag if that helps you. (?)

**Comment by wilhem-meignan** [comment]
Yes, that would be nice.

**Comment by matepek** [comment]
There are some problem with the tests on windows. I have to postpone the release to the beginning of the next week. The attached one can be used until then.

**Comment by wilhem-meignan** [comment]
OK, thank you for the information.

**Comment by wilhem-meignan** [comment]
Update to the current state of the extension: 

> Detecting test cases. Test targets with no test cases are ignored but that's fine with me.

Apparently they are not ignored. I must have not build all the test targets properly. Now I also see empty test suites.

I have found an additional issue: the inline display of failed assertions is not completely reliable. The failed assertions are sometimes displayed but then cleared again immediately. I am not sure what causes them to be cleared.

**Comment by matepek** [comment]
> the inline display of failed assertions is not completely reliable

Yepp, also a know issue. Basically when I list the tests I have no information about the file or line.
After the first run I update it but then the run info is became lost because of the reload. I've made a workaround now, but it would be nice to be fixed. Would you create an issue for doctest and ask them to add some way to list the tests with suite, file and line info?

**Comment by wilhem-meignan** [comment]
Hi @matepek,

I have created an issue for doctest: https://github.com/onqtam/doctest/issues/320

How would you like the command output to look like ideally?

**Comment by matepek** [comment]
xml would be great. `-r=xml`

**Comment by matepek** [comment]
Released it. Since doctest 2.3.6 was released with some improvements I removed some not so stable solution so it does not support lower version of the framework.

Also it lacks some for example grouping by suites, but that requires more time.
The skipped test detection still not supported due to lack of doctest feature.

Have fun with it.

IT will stay "experiment" until there is no tests. (it will take a while)
